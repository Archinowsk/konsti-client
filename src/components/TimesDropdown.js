/* @flow */
import React from 'react'
import { useTranslation } from 'react-i18next'
import { timeFormatter } from 'utils/timeFormatter'
import type { StatelessFunctionalComponent, Element } from 'react'

export type Props = {|
  onChange: Function,
  selectedTime: string,
  times: $ReadOnlyArray<string>,
|}

// TODO: Only enable next open signup
// Check current time and enable new timestamp
// Show "signup starts xx:xx" on others
// Toggle to show upcoming gameslots or all gameslots

export const TimesDropdown: StatelessFunctionalComponent<Props> = (
  props: Props
): Element<'div'> => {
  const { times, onChange, selectedTime } = props
  const { t } = useTranslation()

  const sortedTimes = times.map(sortedTime => {
    const formattedDate = timeFormatter.weekdayAndTime(sortedTime)
    return (
      <option value={sortedTime} key={sortedTime}>
        {formattedDate}
      </option>
    )
  })

  return (
    <div className='times-dropdown'>
      <select onChange={onChange} value={selectedTime}>
        <option value=''>{t('noSignupSelected')}</option>
        {sortedTimes}
      </select>
    </div>
  )
}
