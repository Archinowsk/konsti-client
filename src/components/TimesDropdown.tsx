import React, { FunctionComponent, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import { timeFormatter } from 'utils/timeFormatter';

export interface Props {
  onChange: any;
  selectedTime: string;
  times: readonly string[];
}

// TODO: Only enable next open signup
// Check current time and enable new timestamp
// Show "signup starts xx:xx" on others
// Toggle to show upcoming gameslots or all gameslots

export const TimesDropdown: FunctionComponent<Props> = (
  props: Props
): ReactElement<'div'> => {
  const { times, onChange, selectedTime } = props;
  const { t } = useTranslation();

  const sortedTimes = times.map(sortedTime => {
    const formattedDate = timeFormatter.weekdayAndTime({
      time: sortedTime,
      capitalize: true,
    });
    return (
      <option value={sortedTime} key={sortedTime}>
        {formattedDate}
      </option>
    );
  });

  return (
    <div className='times-dropdown'>
      <select onChange={onChange} value={selectedTime}>
        <option value=''>{t('noActiveTimeSelected')}</option>
        {sortedTimes}
      </select>
    </div>
  );
};
