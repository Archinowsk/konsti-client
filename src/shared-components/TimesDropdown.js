import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { translate } from 'react-i18next';

// TODO: Only enable next open signup
// Check current time and enable new timestamp
// Show "signup starts xx:xx" on others
// Toggle to show upcoming gameslots or all gameslots

const TimesDropdown = props => {
  const { games, t, onChange, date } = props;
  const startTimes = [];

  games.forEach(game => {
    startTimes.push(game.date);
  });

  const sortedTimes = [...new Set(startTimes)].sort();

  const times = sortedTimes.map(sortedTime => {
    const formattedDate = moment.utc(sortedTime).format('DD.M.YYYY HH:mm');
    return (
      <option value={sortedTime} key={sortedTime}>
        {formattedDate}
      </option>
    );
  });

  return (
    <div>
      <select onChange={onChange} value={date}>
        <option>
          {t('selectTime')}
        </option>
        <option value="2000-01-01T23:59:00.000Z">
          {t('noSignupSelected')}
        </option>
        {times}
      </select>
    </div>
  );
};

TimesDropdown.propTypes = {
  games: PropTypes.array.isRequired,
  t: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  date: PropTypes.string.isRequired,
};

export default translate()(TimesDropdown);
