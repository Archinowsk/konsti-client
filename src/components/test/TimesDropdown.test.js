/* @flow */
import React from 'react'
import { shallow } from 'enzyme'
import TimesDropdown from '../TimesDropdown'
import type { Props } from '../TimesDropdown'

describe('TimesDropdown', () => {
  it('should render correctly', () => {
    const props: Props = { games: [], onChange: () => {}, date: '' }
    const component = shallow(<TimesDropdown {...props} />)
    expect(component).toMatchSnapshot()
  })
})
