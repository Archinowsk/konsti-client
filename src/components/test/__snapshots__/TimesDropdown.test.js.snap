// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimesDropdown should render correctly 1`] = `
<TimesDropdown
  i18n={Object {}}
  t={[Function]}
  tReady={true}
/>
`;
